pkg/core/native/notary.go

```diff
@@ -0,0 +1,345 @@
+package native
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+
+	"github.com/nspcc-dev/neo-go/pkg/core/dao"
+	"github.com/nspcc-dev/neo-go/pkg/core/interop"
+	"github.com/nspcc-dev/neo-go/pkg/core/interop/runtime"
+	"github.com/nspcc-dev/neo-go/pkg/core/state"
+	"github.com/nspcc-dev/neo-go/pkg/core/storage"
+	"github.com/nspcc-dev/neo-go/pkg/core/transaction"
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neo-go/pkg/smartcontract"
+	"github.com/nspcc-dev/neo-go/pkg/smartcontract/manifest"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm/stackitem"
+)
+
+// Notary represents Notary native contract.
+type Notary struct {
+	interop.ContractMD
+	GAS   *GAS
+	Desig *Designate
+}
+
+const (
+	notaryName       = "Notary"
+	notaryContractID = -6
```

Comment:
`-100500`. It's better for us to minimize chance of clashing with future official Neo contracts.